Automattic/wp-calypso
# Jetpack Cloud (Agencies): Refactor license issue/assign hook

Resolves `1202619025189113-as-1203377005312989`.
Related to #70101.

## Proposed Changes

* Move the hook for issuing and assigning licenses into its own file.
* Extract several auxiliary functions and hooks from the main hook logic, to separate concerns and improve readability.
* Make the list of product slugs to issue an argument to the hook's callback, rather than a parameter of the hook itself.
* Rename the main hook to `useIssueAndAssignLicenses`, to more accurately describe what it does (i.e., issue and assign product licenses).
* Correct misspellings in a few translated strings and Tracks event names.
* Add/update inline documentation.

## Testing Instructions

_**TL;DR:** Full regression testing of the "issue license(s)" flow in the dashboard._

Run through the following scenarios, verifying that license(s) are issued and assigned correctly, with no unwanted "extra" licenses created, no errors with licenses that you request, and appropriate page redirects when the occasion arises.

* Provision one individual (non-bundle) product license from the dashboard. Pay special attention to the success notification for misspellings and proper bolding of the product name.
* Provision two individual product licenses. Pay special attention to the success notification for proper grammar and bolding of product names.
* Provision three individual product licenses. Pay special attention to the success notification for proper grammar and bolding of product names.
* Provision a bundle. Pay special attention to the list of provisioned licenses afterward and ensure only one license was issued/assigned.

## Pre-merge Checklist


---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - 0-as-1203377005312989
